pkg/matcher/request.go

```diff
@@ -0,0 +1,117 @@
+package matcher
+
+import (
+	"fmt"
+	"net/http"
+	"net/textproto"
+	"strings"
+)
+
+// The Request matcher uses a Value matcher and a Headers matcher to match the path and headers of a http request
+type Request interface {
+	fmt.Stringer
+
+	// Map returns the map correspondence of this instance. The returned value can be
+	// used as an argument to NewRequest to create an identical Request.
+	Map() map[string]string
+
+	// Matches returns true if both the path Value matcher and the Headers matcher in this instance are
+	// matched by the given http.Request.
+	Matches(path string, headers http.Header) bool
+}
+
+type request struct {
+	path    Value
+	headers headers
+}
+
+// NewRequest creates a new Request based on the values of the given map. Aside from http headers,
+// the map may contain one of three special keys.
+//
+//   :path-equal: path will match if equal to the value
+//   :path-prefix: path will match prefixed by the value
+//   :path-regex: path will match it matches the regexp value
+//
+func NewRequest(m map[string]string) (Request, error) {
```

Comment:
This suggests that we can only have one of the `--path-*` flags at a time, do we need to add logic to prevent users from using these flags in combination with each other?

Reply:
I thought about that, but it's tricky because the current extension mechanism doesn't really allow for adding a post-validator that checks the resulting flags and it seemed a bit of an overkill to implement that at this point.

I would like to remove the extension mechanism altogether at some point and move all things dealing with `--http-xxx` over to the proprietary daemon. Perhaps we could defer this improvement until we do that?

Reply:
Yeah that seems reasonable.